drivers/pwm/pwm_stm32.c

```diff
@@ -259,20 +287,49 @@ static int pwm_stm32_pin_set(const struct device *dev, uint32_t pwm,
 
 	channel = ch2ll[pwm - 1u];
 
+	/* in LL_TIM_CC_DisableChannel and LL_TIM_CC_IsEnabledChannel,
+	 * the channel param.could be the complementary one
+	 */
+	if ((flags & PWM_STM32_COMPLEMENTARY_MASK) == PWM_STM32_COMPLEMENTARY) {
+		is_channel = ch2ll_n[pwm - 1u];
+#if !defined(LL_TIM_CHANNEL_CH4N)
+		if (pwm >= 4) {
+#else
+		if (pwm > 4) {
+#endif /* LL_TIM_CHANNEL_CH4N */
```

Comment:
Let's define a STM32_MAX_N_CHANNEL constant and use it here.

Reply:
```
/** Maximum number of complemented timer channels. */
#define STM32_MAX_N_CHANNEL 4
```